libpromises/generic_agent.c

```diff
@@ -762,16 +762,10 @@ void GenericAgentInitialize(EvalContext *ctx, GenericAgentConfig *config)
 
     if (config->agent_specific.agent.bootstrap_policy_server)
     {
-        snprintf(vbuff, CF_BUFSIZE, "%s%cfailsafe.cf", GetInputDir(), FILE_SEPARATOR);
-
-        if (stat(vbuff, &statbuf) == -1)
+        if (CheckAndGenerateFailsafe(GetInputDir(), "failsafe.cf"))
         {
             GenericAgentConfigSetInputFile(config, GetInputDir(), "failsafe.cf");
         }
-        else
```

Comment:
Hmm, after a second look, I believe this part of code changes existing behaviour: Currently, we *always* regenerate `failsafe.cf` when bootstrapping, either it exists or not.